yt/frontends/amrvac/io.py

```diff
@@ -4,15 +4,54 @@
 
 
 """
+import os
 
 import numpy as np
 
+from yt.funcs import ensure_list
 from yt.geometry.selection_routines import GridSelector
 from yt.utilities.io_handler import BaseIOHandler
+from yt.utilities.on_demand_imports import _f90nml as f90nml
 
 from .datfile_utils import get_single_block_field_data
 
 
+def read_amrvac_namelist(parfiles):
+    """Read one or more parfiles, and return a unified f90nml.Namelist object.
+
+    This function replicates the patching logic of MPI-AMRVAC where redundant parameters
+    only retain last-in-line values, with the exception of `&filelist:base_filename`,
+    which is accumulated. When passed a single file, this function acts as a mere
+    wrapper of f90nml.read().
+
+    Parameters
+    ----------
+    parfiles : str, os.Pathlike, byte, or an iterable returning those types
```

Comment:
Is there an iterable that isn't a list that's supported here? If not, I'd be explicit and change this to list.

Reply:
A numpy array or a tuple in particular should work already. Though I can't think of any other useful utterable, I think ALL iterables should be supported already, given the way the first line is written.